complete-firebase-authentication-react-tutorial.md

```diff
@@ -1114,118 +1384,144 @@ class App extends Component {
+Start your application and verify that your sign up, sign in, and sign out functionality works, and that the Navigation component displays the options depending on the session state (authenticated user).
 
-In this section, we will abstract the session handling away with higher order components and React's context API. It has two advantages:
+You have successfully implemented the authentication process with Firebase in React. Everything in the following sections is considered extra, to improve the developer's experience and add a couple of useful features along the way.
 
-* The higher order component fulfils only one purpose. It shields away the business logic from the App component. Thus the App component stays lightweight. There is no business logic mixed up in the component anymore. Higher order components are a great concept in React to extract logic from components, but you can use them later on to enhance components with it. Therefore, they are a great way to accomplish reusability, composability and often maintainability in React.
+### Exercises:
 
-* [React's context API](https://www.robinwieruch.de/react-context-api/) is a React concept which helps us to pass around properties in our application. Rather than passing props explicitly down to all components who are interested in them, you can pass these props implicitly down to these components without bothering the components in between of the hierarchy. Thus, in our case, the App component doesn't need to bother about the authenticated user object anymore, because it only passes it down to various other components.
+* Read more about {{% a_blank "Firebase's Authenticated User" "https://firebase.google.com/docs/auth/web/manage-users" %}}.
+* Confirm your {{% a_blank "source code for the last section" "https://github.com/the-road-to-react-with-firebase/react-firebase-authentication/tree/8fa558878fa5c7e7bbb905c0c12b592cbfd69e3d" %}}.
 
-First, you can revert the recent changes in the App component. It can become a functional stateless component again. It doesn't need to know about the authenticated user object anymore.
+{{% chapter_header "Session Handling with Higher-Order Components" "react-firebase-session-handling-higher-order-components" %}}
 
-In *src/components/App.js* file:
+We added a basic version of session handling in the last section. However,  the authenticated user still needs to be passed down from the App component to interested parties. That ca becomes tedious over time, because the authenticated user has to be passed through all components until it reaches all the component in the leaf components. You used the React Context API to pass down the Firebase instance to any component before. Here, you will do the same for the authenticated user. In a new *src/components/Session/context.js* file, place the following new React Context for the session (authenticated user):
```

Comment:
"it reaches all the component in the leaf components" -> "it reaches all the leaf components"